### PR TITLE
Add support for bus_type nvme

### PR DESCRIPTION
**Pull request by Gnitset**
We're using nvme for our disks and need support to resize them in our deployment.

Right not that is not supported, but this fix should make it work.
It works for us.

### CODE REVIEW

**Review by Didainius — COMMENTED** [review]
Thanks for this PR. 
A few more queries:
* Would you be able to add a CHANGELOG note - create a file in `.changes/v3.3.0` directory named 68-improvements.md and add a line as in similar comments
* Add nvme in docs `vapp_vm.html.markdown` and `vm_internal_disk.html.markdown` files

**Comment by Didainius** [comment]
@Gnitset , 
Thanks for adjustments. I just made a PR for your PR branch (hope this is not too complicated). 
The main reason is we just want to have a test to cover that functionality and I am not sure you have a test env.

https://github.com/Gnitset/terraform-provider-vcd/pull/1

I should be good to go after that, we'll just ask a second look from @dataclouder afterwards

**Review by dataclouder — APPROVED** [review]

File: .changes/v3.3.0/680-improvements.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Add support for disk controller type nvme [GH-680]
```
Comment (Didainius) [inline .changes/v3.3.0/680-improvements.md:1]: Sorry for having this stuck. Could you just merge with master and change .changes to `v3.4.0`. I will just double check after latest merge with master and approve it

Comment (Didainius) [reply .changes/v3.3.0/680-improvements.md:1]: ping

Comment (Gnitset) [reply .changes/v3.3.0/680-improvements.md]: rebased on upstream master and moved .changes to v3.4.0

File: website/docs/r/vapp_vm.html.markdown [hunk 2]
```diff
@@ -360,7 +360,7 @@ Changes are ignored on update. This part isn't reread on refresh. To manage inte
  
 ~> **Note:** Managing disks in VM is possible only when VDC fast provisioned is disabled.
 
-* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`. 
+* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`. **Note** `nvme` is supported in VCD 10.2.1+
```
Comment (dataclouder) [inline website/docs/r/vapp_vm.html.markdown:363]:
```suggestion
* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`. **Note** `nvme` requires *v3.4.0+* and VCD *10.2.1+*
```

File: website/docs/r/vm_internal_disk.html.markdown [hunk 3]
```diff
@@ -43,7 +43,7 @@ The following arguments are supported:
 * `vapp_name` - (Required) The vAPP this VM internal disk belongs to.
 * `vm_name` - (Required) VM in vAPP in which internal disk is created.
 * `allow_vm_reboot` - (Optional) Powers off VM when changing any attribute of an IDE disk or unit/bus number of other disk types, after the change is complete VM is powered back on. Without this setting enabled, such changes on a powered-on VM would fail. Defaults to false.
-* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`. 
+* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`. **Note** `nvme` is supported in VCD 10.2.1+
```
Comment (dataclouder) [inline website/docs/r/vm_internal_disk.html.markdown:46]:
```suggestion
* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`, `nvme`. **Note** `nvme` requires *v3.4.0+* and VCD *10.2.1+*
```

File: vcd/resource_vcd_vm_internal_disk_test.go [hunk 4]
```diff
@@ -363,3 +364,81 @@ resource "vcd_vm_internal_disk" "{{.DiskResourceName}}_ide" {
...
+	// NVM devices are in VCD starting with version 10.2.1
+	client := createTemporaryVCDConnection()
+	if client.Client.APIVCDMaxVersionIs("< 35.1") {
+		t.Skip("NVMe drive support was only introduced in VCD 10.2.1")
+	}
+
+	var params = StringMap{
+		"Org":      testConfig.VCD.Org,
+		"Vdc":      testConfig.VCD.Vdc,
+		"FuncName": "TestVappVmDS",
```
Comment (dataclouder) [inline vcd/resource_vcd_vm_internal_disk_test.go:387]:
```suggestion
		"FuncName": t.Name(),
```

**Review by dataclouder — CHANGES_REQUESTED** [review]
Good job!
Some small changes required

**Comment by Didainius** [comment]
Closing this one in favor of #739 which should include addresses comments.

**Pull request closed without merging**